templates/cluster/vsphere-standalone-cp/templates/vspheremachinetemplate-worker.yaml

```diff
@@ -17,8 +17,11 @@ spec:
             networkName: {{ .Values.worker.network }}
           {{- if .Values.ipamEnabled  }}
             addressesFromPools:
-             - {{- toYaml .Values.ipPool.data | nindent 14 }}
+             - {{- toYaml .Values.ipPool.config | nindent 14 }}
           {{- end }}
+          {{- with .Values.nameservers }}
+            nameservers: {{ . }}
+           {{- end }}
```

Comment:
```yaml
      network:
        devices:
        - dhcp4: {{ not .Values.ipamEnabled }}
          networkName: {{ .Values.worker.network }}
          {{- if .Values.ipamEnabled  }}
          addressesFromPools:
            - {{- toYaml .Values.ipPool.config | nindent 14 }}
          {{- end }}
          {{- with .Values.nameservers }}
          nameservers: {{ toYaml . | nindent 12 }}
          {{- end }}
```

The correct rendering for the `nameservers` along with aligning the template with the control-plane's template (minus 1 tab indent in the `network.devices[]`)